### PR TITLE
modified template arguments

### PR DESCRIPTION
**Pull request by Vidya2606**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ inputs:
     required: false
   namespace:
     description: "Relevant if renderEngine == helm. Namespace to be used for Helm option."
+  arguments:
+    description: "Relevant if renderEngine == helm. Array of arguments. Each argument should be mentioned on a newline"
```
Comment (OliverMKing) [inline action.yml:13]: It's not an array. Specify that it's going to be passed to the eventual helm command

File: src/run.ts [hunk 2]
```diff
@@ -93,10 +93,14 @@ export class HelmRenderEngine extends RenderEngine {
         const args: string[] = [];
         args.push('template');
 
-        const namespace = core.getInput('namespace', { required: false });
-        if (namespace) {
-            args.push('--namespace');
-            args.push(namespace);
+        const allArgs = core.getInput('arguments', { required: false })
```
Comment (OliverMKing) [inline src/run.ts:96]: we aren't passing in all args. we are passing in additional args

File: README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ Sets output variable 'manifestsBundle' which contains the location of the manife
    with:
       renderEngine: 'helm'
       helmChart: './aks-helloworld/'
+      arguments: './aks-helloworld/values-argument.yaml'
```
Comment (OliverMKing) [inline README.md:16]: give a better example. This would run a command that doesn't make any sense.

https://helm.sh/docs/helm/helm_template/

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: action.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ inputs:
   namespace:
     description: "Relevant if renderEngine == helm. Namespace to be used for Helm option."
   arguments:
-    description: "Relevant if renderEngine == helm. Array of arguments. Each argument should be mentioned on a newline"
+    description: "Relevant if renderEngine == helm. Arguments to be passed to the Helm command"
```
Comment (OliverMKing) [inline action.yml:13]: passed to the Helm template command. Add the template thing to be specific

File: src/run.test.ts [hunk 5]
```diff
@@ -136,9 +136,10 @@ describe('Test all functions in run file', () => {
         expect(core.setOutput).toBeCalledWith('manifestsBundle', path.join('tempDirPath', 'baked-template-12345678.yaml'));
     });
 
-    test('HelmRenderEngine() - validate template argument string', async () => {
+    test('HelmRenderEngine() - additional arguments', async () => {
         jest.spyOn(helmUtil, 'getHelmPath').mockResolvedValue('pathToHelm');
         jest.spyOn(core, 'getInput').mockReturnValueOnce('pathToHelmChart').mockReturnValueOnce('releaseName');
+        jest.spyOn(core, 'getInput').mockReturnValueOnce('additionalArguments').mockReturnValueOnce('arguments');
```
Comment (OliverMKing) [inline src/run.test.ts:142]: add unit tests for multiline additional arguments to test expected behavior

**Review by OliverMKing — APPROVED** [review]
Looking good :). Two more small things

File: README.md [hunk 6]
```diff
@@ -13,6 +13,9 @@ Sets output variable 'manifestsBundle' which contains the location of the manife
    with:
       renderEngine: 'helm'
       helmChart: './aks-helloworld/'
+      arguments: | 
+          --ca-file
+          ./ca-file/
```
Comment (OliverMKing) [inline README.md:18]: we should add an example for kustomize as well

File: action.yml [hunk 7]
```diff
@@ -9,6 +9,9 @@ inputs:
     required: false
   namespace:
     description: "Relevant if renderEngine == helm. Namespace to be used for Helm option."
+  arguments:
+    description: "Relevant if renderEngine == helm. Arguments to be passed to the Helm template command"
```
Comment (OliverMKing) [inline action.yml:13]: no mention that it works for kustomize as well here

**Review by OliverMKing — CHANGES_REQUESTED** [review]

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: src/run.ts [hunk 8]
```diff
@@ -87,17 +87,10 @@ export class HelmRenderEngine extends RenderEngine {
...
     }
 
-    private getTemplateArgs(chartPath: string, isV3: boolean): string[] {
+    private async getHelmTemplateArgs(chartPath: string, isV3: boolean): Promise<string[]> {
         const releaseName = core.getInput('releaseName', { required: false });
 
-        const args: string[] = [];
-        args.push('template');
-
-        const namespace = core.getInput('namespace', { required: false });
```
[the remark this answers is not in the document]
Comment (Vidya2606) [reply src/run.ts:96]: I added it back!

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]

**Review by tbarnes94 — APPROVED** [review]

**Pull request merged**